src/profile-logic/flame-graph.js

```diff
@@ -39,29 +44,135 @@ type Stack = Array<{
   nodeIndex: IndexIntoCallNodeTable,
 }>;
 
+/**
+ * Obtain collections of callnode indices needed for building the
+ * flame graph.
+ *
+ * The returned object contains two arrays, one for the roots and one
+ * for all children of the call tree. Along with the children array is
+ * an offset array used to index into it.
+ */
+export function getRootsAndChildren(
```

Comment:
I'm finding myself confused by the returned data structure. What do you think about providing a type for it, and documenting the values there? Specifically a summary of what values (indexes) the data contains, and what determines the sort order. That way it provides a quick summary that is helpful for the more detailed comments below.


e.g. something like the following, but with a little bit more complete comments.
```
type RootsAndChildren = {
  // A list of CallNodeIndexes sorted by XXX, or 0 if XXX.
  children: Uint32Array,
  // The number of elements to slice off for the children, in the same sort order as `children`.
  childrenOffsets: Uint32Array, 
  // A list of every root CallNodeIndex in the call tree, sorted by XXX.
  roots: CallNodeIndex[],
};
```

Reply:
I think that's a great idea! :+1: Thanks